### ISSUE TITLE
import_from_git doesn't correctly pull

### ISSUE THREAD

**Issue by KrauseFx** [KrauseFx/fastlane#539]
The action doesn't seem to correctly pull the changes when it's already cloned. Also it seems like there are too many subfolders when cloning.

**Comment by KrauseFx** [comment]
```
[11:15:44]: [SHELL]: Aborting
[11:15:44]: [SHELL]: Updating a6aa9f4..04fb3d9
[11:15:44]: [SHELL COMMAND]: rm -rf '/tmp/fl_clones/fastlane'
[11:15:44]: [SHELL COMMAND]: git clone 'https://github.com/KrauseFx/fastlane' '/tmp/fl_clones/fastlane' --depth 1 -n
```

**Comment by Liquidsoul** [comment]
Hey @KrauseFx!

I've looked into this and I have performed some `git` tests to make sure why the current implementation behaves this way.

The fact that the `pull` is failing is because of the `-n`. Even if the repository is not checked-out – meaning here the folder starts empty except from `.git` – the git status is dirty because the folder is missing all the _not-checkout_ files. While this does not cause issues when only the `fastlane/Fastfile` or `fastlane/actions/` files are modified, as soon as another file is modified is pulled from the remote, the `check-out` part of the `pull` will fail because the modified file is in `deleted` state.

If I understand your statement `it seems like there are too many subfolders when cloning` correctly,  creating a shallow clone using `--depth 1` does not prevent files and subfolders to be in the clone.
To limit the clone to just some files, fastlane could use a `sparse-checkout` as described in [this conversation](http://stackoverflow.com/questions/600079/is-there-any-way-to-clone-a-git-repositorys-sub-directory-only/13738951#13738951) or in [this tutorial](http://jasonkarns.com/blog/subdirectory-checkouts-with-git-sparse-checkout/).

The bash script to clone the imported repository would look something like this:
```bash
git init /tmp/fl_clone/the_remote
cd /tmp/fl_clone/the_remote
git remote add origin the_remote_url
git config core.sparsecheckout true
echo fastlane/Fastfile >> .git/info/sparse-checkout
echo fastlane/actions >> .git/info/sparse-checkout
git pull origin the_branch
```

This should also fix the issue I mentioned in #513 about changing the value of the `branch` parameter.

As I've already touched this part of the code, I can look into this if you want? I'd be glad to :smiley:
